homeassistant/components/niu/__init__.py

```diff
@@ -0,0 +1,108 @@
+"""The NIU integration."""
+import logging
+
+from niu import NiuAPIException, NiuCloud, NiuNetException, NiuServerException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.event import track_utc_time_change
+import homeassistant.util.dt as dt_util
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, MIN_SCAN_INTERVAL, NIU_COMPONENTS
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): vol.All(cv.positive_int, vol.Clamp(min=MIN_SCAN_INTERVAL)),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
```

Comment:
We no longer allow for new integration to be set up via YAML, instead, we require them to be set up via a configuration flow (the UI).

See: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

For more information on getting started with this, see our developers portal:
<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>